### ISSUE TITLE
Adding Python code

### ISSUE THREAD

**Issue by NimaSarajpoor** [daviddalpiaz/r4sl#22]
Hi David,

This is Nima, and I am almost at the end of my PhD program.

I was searching about something related to Introduction to Statistical Learning, and found your work (pdf file and then I used it to find this repo). Such a great work! 

I have had a similar idea about this book. My idea was to create an interactive environment so that reader can play around the code and try "what if" questions they have in mind. I recently uploaded three of my notebooks here: [blog](https://nimasarajpoor.github.io/blog/) (it might seem a little bit messy and I may go through it later to refine it)

I am using python and I know it is usually harder to get some statistics (e.g. I think regression in `scikit-learn` does not provide p-value or at least I couldn't find an easy way for it there, so I had to use `statsmodels` library. ). My goal is to provide python-code not just for  lab/exercise, but for examples of the book and try to reproduce them. I also provided some links to stackoverflow /youtube videos/ ... when I felt reader might get benefit by getting into some details (e.g. I added a link that explains why we need to z-normalize before PCA).

I was wondering if you have any feedback/suggestion about my work. Also, do you think readers of your book will benefit from my notebooks? Because, to be honest, the format you used for writing the book is very nice and clean. And, it would be really great if I could use your insight on cleaning my notebooks and (if you are interested) adding it to the book. For example, maybe just provide a hyperlink that takes reader to a notebook and they can just play with the code. (or maybe try it on their own and then click on it to see the difference and ....)

Please let me know what you think. Sorry for the long post as I got super excited when I found about this repo :)

**Comment by daviddalpiaz** [comment]
Hi Nima,

First, an overall notes: I'm not really actively maintaining this project at the moment. I actually attempted to completely overhaul it here: https://github.com/daviddalpiaz/bsl Although, I'm also not exactly maintaining that at the moment either.

Next, two brief pieces of advice:

1. Know what resources are already available. ISL is super well known, and there are many add-on tutorial type websites. I assume these have grown in number since I last surveyed this area, especially with the release of the second edition.
2. Know your audience. I'm guessing for you, this might be non-trivial as it looks like you're trying to maintain a general blog. 

For me, these two items are what determines the direction of these projects. Because I have a very particular audience, students in my course, if there is not a resource that I feel properly matches their needs, I create it, which is what happened here. ISL is great, but for the very specific needs of my course, I found myself wanting to "change" ISL here and there, so I started writing my own notes. Tackling a general use-case problem is much more difficult.
